src/systems/pincer.rs

```diff
@@ -0,0 +1,58 @@
+use crate::components::PincerAi::Attacking;
+use amethyst::{
+    core::Transform,
+    ecs::{Join, ReadStorage, System, WriteStorage},
+    renderer::{palette::Srgba, resources::Tint},
+};
+
+use crate::components::{Direction, Motion, Pincer, PincerAi};
+
+pub struct PincerAiSystem;
+
+/// Execute Pincer AI logic.
+/// Attack logic is exceedingly simple. Check the direction of the target and change the sign of
+/// the pincer's velocity to make it run towards the target.
+/// Also changes the pincers color tint depending on whether it's aggro or not.
```

Comment:
Nice documentation. Really helpful.

About the logic, I would like to suggest one change. With the current logic, if the pincer is attacking and the marine tries to evade the attack by jumping over the pincer, the pincer changes direction as soon as the marine crosses over the pincer, which is in mid air. This makes evading a bit too difficult in my opinion.

To make things a bit easier, one way that I can think of is to change the logic so that the pincer doesn't follow the target as long as the target is in air.  So effectively, the pincer would not change its direction when the marine crosses over it and is in air. It would only change its direction once the marine lands on the other side.

Let me know what you think. Feel free to provide any other suggestions / idea.

Reply:
This is definitely a problem. I like your idea; it's nice and simple. One other solution that immediately came to mind is to build in a reaction-time delay in what the pincer does.

As an aside, I noticed something cool when I was testing this. If you go into the code and make the marine invincible, you can jump on the pincer's back and drive it like a surfboard.

Reply:
Reaction-time delay should also work. I am fine with either for now. I would suggest to implement whichever is easier unless you have a preference.

Actually, if you are really careful you can surfboard without making any code changes. The trick is to time the jump precisely enough to land on the pincer's back and then move in small increments. It's an Easter egg. :P

Jokes apart, the marine dies only if it collides with the pincer from the front as per current implementation. I haven't been able to decide how to handle collision from other sides.

Reply:
Alright, I went with the jump-check, because I think it was less work. 

It should hopefully be good to merge now. I rebased the commits on top of master, during the rebase I merged the commits that introduced and fixed the bug, so they now cancel out.

Reply:
Nice work! Just a couple of more comments. We are almost there. :)